lib/matplotlib/colors.py

```diff
@@ -1215,36 +1215,57 @@ def reversed(self, name=None):
 
 class Normalize:
     """
-    A class which, when called, linearly normalizes data into the
-    ``[0.0, 1.0]`` interval.
+    A class which, when called, maps data values within the interval
+    ``[vmin, vmax]`` linearly to fit within the interval ``[0.0, 1.0]``. Data
+    with values outside ``[vmin, vmax]`` will be mapped based on *clip*.
+
+    Examples
+    --------
+    ::
+
+        # [0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0]
+        x = np.linspace(0, 1, 11)
+
+        norm = mpl.colors.Normalize(vmin=0.2, vmax=0.6, clip=False)
+        norm(x) # [-0.5, -0.25, 0, 0.25, 0.5, 0.75, 1.0, 1.25, 1.5, 1.75, 2.0]
+
+        norm = mpl.colors.Normalize(vmin=0.2, vmax=0.6, clip=True)
+        norm(x) # [0, 0, 0, 0.25, 0.5, 0.75, 1.0, 1.0, 1.0, 1.0, 1.0]
```

Comment:
It may be confusing to choose the actual `x` range to be `[0, 1]` which is incidentally the target range of the Norm, but is completely unrelated. These numbers may make the effect more obvious

```suggestion
        x = [-2, -1, 0, 1, 2]
        
        norm = mpl.colors.Normalize(vmin=-1, vmax=1, clip=False)
        norm(x)  # [-0.5, 0., 0.5, 1., 1.5]

        norm = mpl.colors.Normalize(vmin=-1, vmax=1, clip=True)
        norm(x)  # [0., 0., 0.5, 1., 1.]
```